Source/CBL_Router.m

```diff
@@ -573,6 +571,9 @@ - (void) processRequestRanges {
         return;
     }
 
+    if (to < from)
```

Comment:
Agreed. Checking the valid of the range should take precedent. 

What do you think about the case of `500-` (bodyLength=100)? I understand that it is syntactically valid and it would return 416.